packages/adapter-pg/src/conversion.ts

```diff
@@ -129,7 +255,7 @@ export function fieldToColumnType(fieldTypeId: number): ColumnType {
         // Postgres Custom Types
         return ColumnTypeEnum.Enum
       }
-      throw new Error(`Unsupported column type: ${fieldTypeId}`)
+      throw new UnsupportedNativeDataType(fieldTypeId)
```

Comment:
We could make this function return a result too for consistency and so we don't need to catch an error and convert it to a result later, but let's better merge this as is. I have some relevant changes and a helper function added in my PR and it will be easier to integrate and refactor it there.